### PR TITLE
AI Chatbot

### PR DESCRIPTION
**Pull request by Steve-KX-RL**
1. Have a (somewhat) useable conversation with your Azure AI chatbot, after inputting the credentials.
2. Focus camera onto tools when clicking the tool name in chat history
3. Top menu bar --> Help --> Show AI Chatbot to show/hide chatbot
4. Also need edge-server and user-interface to work.

### CODE REVIEW

File: content_scripts/VirtualCamera.js [hunk 1]
```diff
@@ -294,7 +294,8 @@ import * as THREE from '../../thirdPartyCode/three/three.module.js';
             document.addEventListener('pointermove', function (event) {
                 this.mouseInput.latest.x = event.pageX;
                 this.mouseInput.latest.y = event.pageY;
-                if (this.mouseInput.isRotateRequested || this.mouseInput.isStrafeRequested) return;
+                // if (this.mouseInput.isRotateRequested || this.mouseInput.isStrafeRequested) return;
```
Comment (hobinjk-ptc) [inline content_scripts/VirtualCamera.js:297]: Can just delete this line

File: content_scripts/desktopAdapter.js [hunk 2]
```diff
@@ -139,6 +139,14 @@ createNameSpace('realityEditor.device.desktopAdapter');
                     color: env.groundWireframeColor,
                     thickness: 0.075 // relatively thick
                 });
+                // todo Steve: make ai chatbox turn on dark mode
+                let aiContainer = document.getElementById('ai-chat-tool-container');
+                let searchTextArea = [...aiContainer.querySelectorAll('#searchTextArea')][0];
```
Comment (hobinjk-ptc) [inline content_scripts/desktopAdapter.js:144]: This is equivalent to the more straightforward `aiContainer.querySelector('#searchTextArea')`

**Review by hobinjk-ptc — APPROVED** [review]

File: content_scripts/setupMenuBar.js [hunk 3]
```diff
@@ -99,6 +101,9 @@ createNameSpace('realityEditor.gui');
 
         const toggleFlyMode = new MenuItem(ITEM.ToggleFlyMode, { toggle: true, shortcutKey: 'F', defaultVal: false }, null);
         menuBar.addItemToMenu(MENU.Camera, toggleFlyMode);
+        
+        const focusCamera = new MenuItem(ITEM.FocusCamera, { shortcutKey: 'G' }, null);
+        menuBar.addItemToMenu(MENU.Camera, focusCamera);
```
Comment (benptc) [inline content_scripts/setupMenuBar.js:106]: I'm a little bit double-minded about this one. On one hand, it's nice to have it visible in the menu so you know that the G keyboard shortcut exists. On the other hand, if you click on the menu item and the mesh isn't underneath your mouse position when you focus, it seems to focus onto nothing (just seeing a grey screen).

**Review by benptc — APPROVED** [review]
Mostly LGTM, just debating how to encourage using the keyboard instead of clicking on the menu shortcut for focusing the camera.

**Pull request merged**